frontend/src/common/components/formInputField/FormOwnershipInputField.tsx

```diff
@@ -172,14 +168,15 @@ export default function FormOwnershipInputField({
     }, [formData.identifier, setIsInvalidSSNAllowed]);
 
     useEffect(() => {
-        if (!isCreating && !createError && createData) {
+        if (!isCreating && !createError && createData && doesAContainB(createData, formData)) {
+            setFormData({name: "", identifier: "", email: ""});
             hitasToast("Omistaja lisätty onnistuneesti!");
-            setDisplayedValue(createData.name);
+            setDisplayedValue(`${createData.name} (${createData.identifier})`);
             setFieldValue(createData.id);
             setIsAddingNew(false);
             closeModal();
         }
-    }, [createData, createError, isCreating, setFieldValue]);
+    }, [createData, createError, isCreating, setFieldValue, formData, setFormData]);
```

Comment:
I'd see that as more of a bug ticket of its own, as it's not directly a blocking thing for the sales page functionality to be merged. It will be remedied by the front end validation, too - as it will be impossible to create a user like that. :)